src/com/oltpbenchmark/GeoPartitionedConfigFileOptions.java

```diff
@@ -0,0 +1,120 @@
+package com.oltpbenchmark;
+
+import java.util.Optional;
+
+import org.apache.commons.configuration.ConfigurationException;
+
+import com.oltpbenchmark.util.GeoPartitionPolicy;
+import com.oltpbenchmark.util.InvalidUserConfiguration;
+import com.oltpbenchmark.util.PlacementBlock;
+import com.oltpbenchmark.util.PlacementPolicy;
+
+public class GeoPartitionedConfigFileOptions extends ConfigFileOptionsBase {
+    
+    public GeoPartitionedConfigFileOptions(String filePath) throws ConfigurationException {
+        super(filePath);
+    }
+    
+    private boolean isGeoPartitioningEnabled() {
+        return isFlagSet("enableGeoPartitionedWorkload");
+    }
+
+    private Optional<Integer> getNumberOfPartitions() {
+        return getIntOpt("numberOfPartitions");
+    }
+
+    private int getNumTablespaces() {
+        return xmlConfig.getList("tablespaces/tablespace/name").size();
+    }
+
+    public GeoPartitionPolicy getGeoPartitionPlacement(final int totalWarehousesAcrossShards, int numWarehouses) {
+        // First check whether partitioning enabled.
+        if (!isGeoPartitioningEnabled()) {
+            // Partitioning is disabled. No point setting or looking for any other options.
+            return null;
+        }
+        
+        // Verify that the total number of warehouses can be evenly divided across partitions.
+        final int numPartitions = getNumberOfPartitions().get();
+        if (numPartitions * numWarehouses != totalWarehousesAcrossShards) {
```

Comment:
Hey @deeps1991 , we support multiple clients for tpcc and that is why we had totalWarehousesAcrossShards and startWarehouseIdForShard. A client does not know about other clients and it knows only its own startWarehouseIdForShard and numWarehouses.

It does not look like that we are supporting multiple clients per geography. However, the usage of numWarehouses and totalWarehousesAcrossShards in context of geo partitioning has made the things confusing. Two comments here:
1. If multiple clients are not supported for a single geo partitioned region (which I think should be. It is a matter of time before @rkarthik007 would need 100K WH run on geo partitioned cluster), we should throw an exception during config validation.  
2. Use different variable for number of Warehouses in the current partition so that we can add support for 1 in future.

Reply:
@hbhanawat The way I thought about it is, every client connects to its own partition. 
So let's say you have 100K warehouses and you want to have (just as an example) 20 clients in total. Then you can create 20 partitions, and each partition will house 5K warehouses. You can spread these clients across geographies however you want, so I'm not completely sure what you mean when you say that multiple clients are not supported for single geo partitioned run.
Am I misunderstanding something here? AFAIK totalWarehousesAcrossShards would be 100K, numWarehouses=5k and numPartitions=20 (and total number of clients =20) in the above example. If you have geo distributed cluster of 2 regions you can 10 partitions/clients in each region, or is something wrong above.
Do you mean to say that we should not tie number of clients with number of partitions?